src/libraries/System.Data.Common/tests/System/Data/SqlTypes/SqlXmlTest.cs

```diff
@@ -119,6 +96,82 @@ public void SqlXml_fromXmlReader_CreateReaderTest()
             Assert.Equal(xmlStr, xrdr.ReadOuterXml());
         }
 
+        [Theory]
+        [InlineData("element_whitespace-text.xml")]
+        [InlineData("root_qname.xml")]
+        [InlineData("sample_ecommerce.xml")]
+        [InlineData("sql_batch_request.xml")]
+        [InlineData("sql_batch_response.xml")]
+        [InlineData("sql_datatypes-1.xml")]
+        [InlineData("sql_datatypes-2.xml")]
+        [InlineData("sql_datatypes-3.xml")]
+        [InlineData("xmlns-1.xml")]
+        [InlineData("xmlns-2.xml")]
+        [InlineData("xmlns-3.xml")]
+        [InlineData("xmlns-4.xml")]
+        [InlineData("comments_pis.xml")]
+        [InlineData("element_content_growth.xml")]
+        [InlineData("element_nested-1.xml")]
+        [InlineData("element_nested-2.xml")]
+        [InlineData("element_nested-3.xml")]
+        [InlineData("element_single.xml")]
+        [InlineData("element_stack_growth.xml")]
+        [InlineData("element_tagname_growth.xml")]
+        [InlineData("element_types.xml")]
+        [InlineData("element_whitespace-modes.xml")]
+        public void SqlXml_fromXmlReader_TextXml(string filename)
+        {
+            string filepath = Path.Combine("TestFiles/SqlXml/TextXml", filename);
+
+            using FileStream xmlStream = new FileStream(filepath, FileMode.Open);
+            SqlXml sqlXml = new SqlXml(xmlStream);
+
+            // Reading XML stored as SQL Binary XML will result in using
+            // the XmlTextReader implementation
+            using XmlReader sqlXmlReader = sqlXml.CreateReader();
+
+            // Read to the end to verify no exceptions are thrown
+            while(sqlXmlReader.Read());
+        }
+
+        [Theory]
+        [InlineData("element_whitespace-text.bmx")]
+        [InlineData("root_qname.bmx")]
+        [InlineData("sample_ecommerce.bmx")]
+        [InlineData("sql_batch_request.bmx")]
+        [InlineData("sql_batch_response.bmx")]
+        [InlineData("sql_datatypes-1.bmx")]
+        [InlineData("sql_datatypes-2.bmx")]
+        [InlineData("sql_datatypes-3.bmx")]
+        [InlineData("xmlns-1.bmx")]
+        [InlineData("xmlns-2.bmx")]
+        [InlineData("xmlns-3.bmx")]
+        [InlineData("xmlns-4.bmx")]
+        [InlineData("comments_pis.bmx")]
+        [InlineData("element_content_growth.bmx")]
+        [InlineData("element_nested-1.bmx")]
+        [InlineData("element_nested-2.bmx")]
+        [InlineData("element_nested-3.bmx")]
+        [InlineData("element_single.bmx")]
+        [InlineData("element_stack_growth.bmx")]
+        [InlineData("element_tagname_growth.bmx")]
+        [InlineData("element_types.bmx")]
+        [InlineData("element_whitespace-modes.bmx")]
+        public void SqlXml_fromXmlReader_SqlBinaryXml(string filename)
+        {
+            string filepath = Path.Combine("TestFiles/SqlXml/SqlBinaryXml", filename);
```

Comment:
This test looks identical except for the path being supplied here?  Should it instead just be the same test and choose a different path based on the extension?  Or just include this part of the path as part of each InlineData?